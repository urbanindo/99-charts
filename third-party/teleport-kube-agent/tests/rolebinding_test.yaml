suite: RoleBinding
templates:
  - rolebinding.yaml
tests:
  - it: Create a RoleBinding when upgrading
    release:
      isupgrade: true
    set:
      unitTestUpgrade: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: RoleBinding

  - it: creates a RoleBinding
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: RoleBinding
      - matchSnapshot: {}

  - it: sets RoleBinding labels when specified
    values:
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: rolebinding
      - matchSnapshot: {}
