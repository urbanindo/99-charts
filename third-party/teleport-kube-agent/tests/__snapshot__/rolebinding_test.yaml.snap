creates a RoleBinding:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: RELEASE-NAME
    subjects:
    - kind: ServiceAccount
      name: RELEASE-NAME
      namespace: NAMESPACE
sets RoleBinding labels when specified:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/name: teleport-kube-agent
        resource: rolebinding
      name: RELEASE-NAME
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: RELEASE-NAME
    subjects:
    - kind: ServiceAccount
      name: RELEASE-NAME
      namespace: NAMESPACE
